django/db/backends/mysql/introspection.py

```diff
@@ -217,14 +217,18 @@ def get_constraints(self, cursor, table_name):
             """
             cursor.execute(type_query, [table_name])
             for constraint, check_clause in cursor.fetchall():
-                constraints[constraint] = {
-                    'columns': self._parse_constraint_columns(check_clause, columns),
-                    'primary_key': False,
-                    'unique': False,
-                    'index': False,
-                    'check': True,
-                    'foreign_key': None,
-                }
+                if constraint not in constraints:
+                    constraints[constraint] = {
+                        'columns': OrderedSet(),
+                        'primary_key': False,
+                        'unique': False,
+                        'index': False,
+                        'check': True,
+                        'foreign_key': None,
+                    }
+                constraints[constraint]['check'] = True
+                for column in self._parse_constraint_columns(check_clause, columns):
+                    constraints[constraint]['columns'].add(column)
```

Comment:
This changes looks unrelated and all tests pass without them.

Reply:
@felixxm, no doubt tests pass for MySQL, but that fails for MariaDB. On debugging I found that `UNIQUE` constraint was removed on `code` column that afterward leads to Foreign key constraint. 

This change is done purposefully. `UNIQUE` constraint is overridden by `check` constraint. This change will merge both constraints thus afterward I can filter `UNIQUE` constraints.

Reply:
:thinking: All tests pass on MariaDB for me.

Reply:
If it is possible that `unique` and `check` constraints return by different queries have the same name on `MariaDB ` we should fix this separately. I can dig into it.

Reply:
@felixxm but that was failing on Jenkins too. (I'm sorry I can't find the history of old failed test reports for reference).

Reply:
Yea I'm confident that was the case. Anyway keep me updated about your findings.

Reply:
I added PR #11704.

Reply:
Merged.

Reply:
Thanks @felixxm